### PR TITLE
[bitnami/mysql] add networkpolicy for primary (#6720)

### LINKED ISSUE #6720: NetworkPolicy for mysql
**Which chart**:
mysql

**Describe the solution you'd like**
Similar to the [postgresql chart's](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/templates/networkpolicy.yaml), would it be possible to implement a policy for this chart too?

Let me know, as I'm happy to attempt creating a merge request ;)

Thanks!

### PR DESCRIPTION
**Pull request by bynare**
**Description of the change**

This changes adds the ability to enable and configure a Network Policy on the primary.

**Benefits**

Better security.

**Possible drawbacks**
n/a

**Applicable issues**

  - fixes #6720

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/mysql/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            matchLabels:
+              {{ template "common.names.fullname" . }}-client: "true"
+          {{- if .Values.primary.networkPolicy.explicitNamespacesSelector }}
+          namespaceSelector:
+{{ toYaml .Values.primary.networkPolicy.explicitNamespacesSelector | indent 12 }}
+          {{- end }}
+        - podSelector:
+            matchLabels:
+            {{- include "common.labels.matchLabels" . | nindent 14 }}
+              role: read
```
Comment (migruiz4) [inline bitnami/mysql/templates/networkpolicy.yaml:33]: The MySQL replicas do not have a label `role: read`. Instead they have `app.kubernetes.io/component: secondary`.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @bynare,

Thank you very much for your contribution! 

I have suggested some changes, could you please have a look?

**Comment by bynare** [comment]
Hey @migruiz4, 

Thanks for the review! :)

I've amended the things you've pointed out - thanks!

**Comment by migruiz4** [comment]
Thank you very much for applying the suggestions!

There is one more thing I'd like to discuss with you. Providing a NetworkPolicy that only applies to the primary nodes seems incomplete.

For example, without a Network policy for secondary replicas, metrics wouldn't be accessible for all the pods.

We should cover both the primary node and secondary replicas with a single networkPolicy. Rename `primary.networkPolicy.*` as `networkPolicy.*` and remove "app.kubernetes.io/component: primary" from the networkPolicy podSelector.

Or otherwise, add a networkPolicy for the secondary replicas (`secondary.networkPolicy.*`) and rename the template `networkpolicy.yaml` to `primary/networkpolicy.yaml`.

Both approaches would be fine for me, feel free to let me know what you think.

**Comment by bynare** [comment]
All good @migruiz4 - I was in two minds initially too: whether to include secondary in this.

> We should cover both the primary node and secondary replicas with a single networkPolicy. Rename primary.networkPolicy.* as networkPolicy.* and remove "app.kubernetes.io/component: primary" from the networkPolicy podSelector.

^ I've made changes to go with this option you suggested ;)

Thanks again for reviewing and bearing with me!

**Review by migruiz4 — APPROVED** [review]
Looks good to me!

Thank you for your contribution!

**Pull request merged**